posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,180 @@
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Continuous integration and delivery (CI/CD) of cloud-native Java applications using Jenkins
+Shamjith Antholi <https://github.com/shamjithantholi>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+[#Intro]
+
+Continuous integration and continuous deployment (CI/CD) is a process in which developers frequently merge code changes into a source code repository, which then automatically starts a code build, security scanning, and test running, and deployment of the application. Automation of this process, through adopting a DevOps culture, tools, and processes, enables organizations to release on a more frequent basis without compromising on quality. There are various tools available to automate the CI/CD process that can be readily integrated into a DevOps deployment pipeline. In this blog post we'll discuss how you can use Jenkins to automate the CI/CD process for building, testing, and deploying your cloud-native Java applications running on link:https://openliberty.io[OpenLiberty].
+
+Jenkins is a popular open source automation server that has hundreds of plugins to help integrate it into your existing DevOps pipeline. For example, Jenkins can pull code changes from a git repository, build the code using Maven, then run Junit tests against the built code, and deploy the application into Kubernetes.
+
+The following diagram shows a simple architecture for a pipeline that builds and deploys a Jakarta EE & MicroProfile application in a single Kubernetes environment. When a code change is detected by Jenkins, Jenkins builds the code, runs security and static analysis scans, and runs unit tests against the built code. Jenkins then generates a Docker image of the application with its Open Liberty server and saves the image to a Docker repository, Jenkins then trigger the deployment using CLI commands and deploy the Docker image to Kubernetes..
+
+image::/img/blog/liberty-devops-generic-architecture-2.png[Liberty DevOps architecture diagram ,width=70%,align="center"]
+
+I will assume that you have a basic understanding of git, Docker, Kubernetes. In this blog post, I will assume that the Jakarta EE & MicroProfile application code are stored in link:https://github.com/[Github]. I will use link:https://hub.docker.com/[DockerHub] to store the containerized application (correction: Docker images)and I will deploy the containerized application to link:https://cloud.ibm.com/kubernetes/catalog/create[IBM Cloud Kubernetes Service].
+
+== Installing and configuring Jenkins to setup CI/CD of Jakarta EE & MicroProfile application
+
+Install *jenkins* with all the recommended plug-ins link:https://www.jenkins.io/doc/book/installing/[install Jenkins], make sure that the following plug-ins are installed. You can install Jenkins on any physical or virtual servers, or it can be running as a container on Kubernetes itself.
```

Comment:
Reword to "Install link:[https://www.jenkins.io/doc/book/installing/[Jenkins]](https://www.jenkins.io/doc/book/installing/%5BJenkins%5D) with the following plugins on the base image of your Jenkins container image:"

Reply:
Replaced with "Install link:https://www.jenkins.io/doc/book/installing/[Jenkins] with the following plugins on the base image of your Jenkins container image, or on the Jenkins master."